File: DSA/PriyankaSingh_590016827/day19question3.c
#include <stdio.h>
#include <stdbool.h>

bool searchMatrix(int rows, int cols, int matrix[rows][cols], int target) 
{
    int row = 0;
    int col = cols - 1;

    while (row < rows && col >= 0) 
    {
        if (matrix[row][col] == target)
            return true;
        else if (matrix[row][col] > target)
            col--;  // Move left
        else
            row++;  // Move down
    }

    return false;
}

int main() 
{
    int rows, cols, target;

    printf("Enter number of rows: ");
    scanf("%d", &rows);

    printf("Enter number of columns: ");
    scanf("%d", &cols);

    int matrix[rows][cols];

    printf("Enter elements of the matrix (row-wise):\n");
    for (int i = 0; i < rows; i++)
        for (int j = 0; j < cols; j++)
            scanf("%d", &matrix[i][j]);

    printf("Enter the target value to search for: ");
    scanf("%d", &target);

    if (searchMatrix(rows, cols, matrix, target))
        printf("Target %d found in the matrix.\n", target);
    else
        printf("Target %d not found in the matrix.\n", target);

    return 0;
}

File: DSA/pranavdixit_590014418/Day_19_Question_3.py
def search_matrix(matrix, target):
    if not matrix or not matrix[0]:
        return False
    row, col = 0, len(matrix[0]) - 1
    while row < len(matrix) and col >= 0:
        if matrix[row][col] == target:
            return True
        elif matrix[row][col] > target:
            col -= 1
        else:
            row += 1
    return False

print(search_matrix([[1,4,7],[2,5,8],[3,6,9]], 5))
print(search_matrix([[1,4,7],[2,5,8],[3,6,9]], 13))
print(search_matrix([[1,3,5],[2,4,6],[7,8,9]], 6))
